src/core/renderers/webgl/managers/FilterManager.js

```diff
@@ -106,7 +106,7 @@ export default class FilterManager extends WebGLManager
 
             // TODO we should fit the rect around the transform..
         }
-        else
+        else if (filters[0].fitToDestination)
```

Comment:
Look at 10 lines above - same strategy applies for other parameters, like resolution. I tried to make working solution without that hack, but so far i failed.